src/Splat.Microsoft.Extensions.Logging/MicrosoftExtensionsLoggingLogger.cs

```diff
@@ -0,0 +1,115 @@
+            new KeyValuePair<LogLevel, global::Microsoft.Extensions.Logging.LogLevel>(LogLevel.Error, global::Microsoft.Extensions.Logging.LogLevel.Error),
+            new KeyValuePair<LogLevel, global::Microsoft.Extensions.Logging.LogLevel>(LogLevel.Fatal, global::Microsoft.Extensions.Logging.LogLevel.Critical)
+        };
+
+        private static readonly ImmutableDictionary<LogLevel, global::Microsoft.Extensions.Logging.LogLevel> _mappingsDictionary = _mappings.ToImmutableDictionary();
+
+        private readonly global::Microsoft.Extensions.Logging.ILogger _inner;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MicrosoftExtensionsLoggingLogger"/> class.
+        /// </summary>
+        /// <param name="inner">The actual Microsoft.Extensions.Logging logger.</param>
+        /// <exception cref="ArgumentNullException">Microsoft.Extensions.Logging logger not passed.</exception>
+        public MicrosoftExtensionsLoggingLogger(global::Microsoft.Extensions.Logging.ILogger inner)
+        {
+            _inner = inner ?? throw new ArgumentNullException(nameof(inner));
+            SetLogLevel();
+        }
+
+        /// <inheritdoc />
+        public LogLevel Level
+        {
+            get; private set;
+        }
+
+        /// <inheritdoc />
+        public void Write(string message, LogLevel logLevel)
+        {
+            if ((int)logLevel < (int)Level)
+            {
+                return;
+            }
+
+            _inner.Log(_mappingsDictionary[logLevel], message);
+        }
+
+        /// <inheritdoc />
+        public void Write(Exception exception, string message, LogLevel logLevel)
+        {
+            if ((int)logLevel < (int)Level)
+            {
+                return;
+            }
+
+            _inner.Log(_mappingsDictionary[logLevel], exception, message);
+        }
+
+        /// <inheritdoc />
+        public void Write(string message, Type type, LogLevel logLevel)
+        {
+            if ((int)logLevel < (int)Level)
+            {
+                return;
+            }
+
+            _inner.Log(_mappingsDictionary[logLevel], $"{type.Name}: {message}");
+        }
+
+        /// <inheritdoc />
+        public void Write(Exception exception, string message, Type type, LogLevel logLevel)
+        {
+            if ((int)logLevel < (int)Level)
+            {
+                return;
+            }
+
+            _inner.Log(_mappingsDictionary[logLevel], exception, $"{type.Name}: {message}");
+        }
+
+        /// <summary>
+        /// Works out the log level.
+        /// </summary>
+        /// <remarks>
+        /// This was done so the Level property doesn't keep getting re-evaluated each time a Write method is called.
+        /// </remarks>
+        private void SetLogLevel()
+        {
+            foreach (var mapping in _mappings)
```

Comment:
hmm.. They don't have any call back and seems to be very log basis if this can change every call. eg the console loggers have:
```
public Func<string, LogLevel, bool> Filter
        {
            get { return _filter; }
            set
            {
                if (value == null)
                {
                    throw new ArgumentNullException(nameof(value));
                }

                _filter = value;
            }
        }

public bool IsEnabled(LogLevel logLevel)
        {
            if (logLevel == LogLevel.None)
            {
                return false;
            }

            return Filter(Name, logLevel);
        }
```

Also other loggers are even more complex.

One consistent thing they are doing is calling "IsEnabled()" every log call anyway (this doesn't seem like the most efficient logging framework out there), so maybe for this framework we should call it every Write, or add a re-evaluate method or something? At the very least I guess we should document if we don't call it every Write.

Reply:
sorry I will take a look